tests/ballerina-spec-conformance-tests/src/test/java/org/ballerinalang/test/TestRunnerUtils.java

```diff
@@ -0,0 +1,263 @@
+            } else if (Pattern.matches(pattern, header)) {
+                if (key != null) {
+                    headers.put(key, value);
+                }
+                int index = line.indexOf(COLON);
+                key = header.substring(0, index).trim();
+                value = header.substring(index + 1).trim();
+                if (key.equals(LABELS)) {
+                    String[] labels = value.split("\\s*,\\s*");
+                    checkLabelsDefined(labels);
+                    kind = isSkippedTestCase(selectedLabels, labels, kind);
+                }
+            } else {
+                value = value + " " + header.trim();
+            }
+        }
+        return kind;
+    }
+
+    private static void checkLabelsDefined(String[] labels) {
+//        for (String label : labels) {
+//            if (!org.ballerinalang.test.Labels.LABELS.contains(label)) {
+//                Assert.fail(String.format("Lable %s is not defined", label));
+//            }
+//        }
+    }
+
+    private static String isSkippedTestCase(Set<String> selectedLabels, String[] labels, String kind) {
+        if (selectedLabels.isEmpty()) {
+            return kind;
+        }
+        for (String label : labels) {
+            if (selectedLabels.contains(label)) {
+                return SKIPPED_TEST;
+            }
+        }
+        return kind;
+    }
+
+    public static String validateError(CompileResult result, List<String> outputValues, String kind,
+                                       List<Integer> errLines, String fileName, int absLineNum, String details) {
+        for (int i = 0; i < result.getDiagnostics().length; i++) {
+            totalTest = totalTest + 1;
+            Diagnostic diagnostic = result.getDiagnostics()[i];
+            int actualLineNum = diagnostic.location().lineRange().startLine().line() + 1 + absLineNum;
+            int expLineNum = errLines.get(i) + absLineNum;
+            boolean output = actualLineNum == expLineNum;
+            if (output) {
+                totalTestPassed = totalTestPassed + 1;
+            }
+            details = details + ReportGenerator.generateErrorDetails(fileName, kind, String.valueOf(actualLineNum),
+                                                          outputValues.get(i), diagnostic.message(), getResult(output));
+            detailsOfTests = detailsOfTests + ReportGenerator.generateTestDetails(fileName, kind,
+                                                                         String.valueOf(expLineNum), getResult(output));
+
+            Assert.assertEquals(actualLineNum, expLineNum, String.format("In %s, incorrect line number:", fileName));
+        }
+        return details;
+    }
+
+    public static void validateOutput(String fileName, List<String> outputValues, String[] results) {
+        if (outputValues.size() != results.length) {
+            Assert.fail(String.format("In %s file, Expected %s but found %s", fileName, outputValues,
```

Comment:
```suggestion
            Assert.fail(String.format("In %s file, expected %s but found %s", fileName, outputValues,
```

Let's update wherever applicable.